src/LLVM_Runtime_Linker.cpp

```diff
@@ -809,11 +809,11 @@ std::unique_ptr<llvm::Module> get_initial_module_for_target(Target t, llvm::LLVM
                 modules.push_back(get_initmod_posix_io(c, bits_64, debug));
                 modules.push_back(get_initmod_linux_host_cpu_count(c, bits_64, debug));
                 modules.push_back(get_initmod_linux_yield(c, bits_64, debug));
-                if (t.has_feature(Target::WasmThreads)) {
-                    modules.push_back(get_initmod_posix_threads(c, bits_64, debug));
-                } else {
-                    modules.push_back(get_initmod_fake_thread_pool(c, bits_64, debug));
-                }
+                // With recent versions of Emscripten, it appears we can use the posix-thread-pool
+                // always, even if `wasm_threads` is not enabled. Will need some more testing
+                // to be sure, but this change greatly simplifies linking together of multiple
+                // wasm modules if they have mixed target strings (some with threads, some without).
+                modules.push_back(get_initmod_posix_threads(c, bits_64, debug));
```

Comment:
So what does wasm_threads even do as a target feature then? If I leave it off and use Func::parallel, what happens?

Reply:
It enables extra compiler flags in the Wasm backend. Which...uh... huh, that's a good question, I don't think there's anything else in Halide's code that stops it from trying to run parallel jobs, so, something is wonky. Let me investigate further.